### ISSUE TITLE
no-unused-vars: varsIgnorePattern doesn't work

### ISSUE THREAD

**Issue by fab1an** [eslint/eslint#3640]
I have `no-unused-vars": [2, {"varsIgnorePattern": "^L$"}],` set to ignore my Logger declaration in each file: `const L = Logger.getLogger("ProPage2.Mutations")`.

It does not work. I'm using eslint 1.0.0

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by BYK** [comment]
Hi @fab1an. This feature was added in ESLint v1.3.0: http://eslint.org/blog/2015/08/eslint-v1.3.0-released/

**Comment by justin-calleja** [comment]
Hi,

I'm extending a base eslint config in my .eslintrc and have overridden "no-unused-vars" with the pattern in the [doc](http://eslint.org/docs/rules/no-unused-vars#varsignorepattern):

```
  "rules": {
    "no-unused-vars" : [1, { "varsIgnorePattern": "[iI]gnored" }]
  }
```

I'm using:

```
./node_modules/.bin/eslint -v
v3.1.1
```

But this code is still giving me a warning:

```js
  child.on('exit', function(e, ignored) {
    console.log('finished');
  });
```

```
/Users/justin/file-based-drill/drill/utils/tmp.js
  10:32  warning  'ignored' is defined but never used  no-unused-vars
```

Also, the pattern I'm looking for is "anything which starts with an underscore". Would this do it: "^\_.+"

Thanks

**Comment by mysticatea** [comment]
http://eslint.org/docs/rules/no-unused-vars#argsignorepattern

**Comment by justin-calleja** [comment]
@mysticatea thanks! (haven't tried it out - I'll do that later. Didn't know there was an argsIgnorePattern in addition to a varsIgnorePattern).

**Comment by justin-calleja** [comment]
Works as advertised - thanks!
